src/main/java/com/obones/binding/airzone/internal/api/model/AirZoneHvacZone.java

```diff
@@ -141,7 +141,11 @@ public int getMode() {
     };
 
     public int[] getSpeeds() {
-        return speeds;
+        int[] value = new int[speeds + 1];
```

Comment:
I think that if `speeds` is not in the source JSON then its value is set to `0` thus leading to a `1` sized array being returned.
As such, I believe that `speeds` should a `@Nullable Integer` like `air_demand` and a few others.
This in turn requires to check for its nullness here and return an empty array if it is `null`.
Then the code would most likely look like this:

```
        if (speeds != null) {
            int[] value = new int[speeds + 1];
            for (int i = 0; i < value.length; i++) {
                value[i] = i;
            }
            return value;
        }
        else {
            return new int[0];
        }
```

Reply:
Got it. Updated with a slight modification, semantically the same.

Reply:
Sorry for the delays, got hooked up on other things.
For a reason beyond me, in VSCode, your version leads to a nullness warning on the access to `speed` inside the array size initializer.
I don't really get why it's different with my `if (speeds != null) ... else` but to see it, you have to declare `speeds` like this:

    private @Nullable Integer speeds;